tests/testthat/test-class-Megastudy.R

```diff
@@ -359,7 +406,15 @@ test_that("imputeZeroes method is sane", {
       variableSpec = new("VariableSpec", variableId = 'pathogen_presence', entityId = 'assay'),
       plotReference = new("PlotReference", value = 'facet1'),
       dataType = new("DataType", value = 'STRING'),
-      dataShape = new("DataShape", value = 'CATEGORICAL'))
+      dataShape = new("DataShape", value = 'CATEGORICAL')),
+    new("VariableMetadata",
+      variableClass = new("VariableClass", value = 'native'),
+      variableSpec = new("VariableSpec", variableId = 'sex', entityId = 'sample'),
+      # empty plotReference means that it is not plotted
+      dataType = new("DataType", value = 'STRING'),
+      dataShape = new("DataShape", value = 'CATEGORICAL'),
+      weightingVariableSpec = VariableSpec(variableId='specimen_count',entityId='sample'),
+      hasStudyDependentVocabulary = TRUE)
```

Comment:
What's the test starting at line 383 doing? There's no comment and my spot-the-difference powers are weak!

Reply:
the one at 341 tests the case where we have a special vocab, a downstream entity and NO specimen_count in the plot. the one at 383 tests the case where we have a special vocab, a downstream entity and also specimen_count in the plot. we said in these cases we would NOT impute zeroes. in the first case (341) bc the variable to impute zeroes on isnt present and in the second case (383) bc the values would be removed later by our requirement for complete cases in the map. if/when the map supports representations of missingness properly, well have to change this test and add support.